### PR TITLE
Sign up feature

### PR DESCRIPTION
**Pull request by jdalto**
**Description:**

This PR adds:
- Sign up and sign in screens.
- Sign up and sign in forms.
- Common components.
- User api.
- User reducers and actions.
- Constraints for sign in and sign up forms.

**Platforms:**
- Android.
- IOS.

**Notes:**

In progress:
- Sign in screen style.
- Sign in validations.
- Handle errors of sign in brought from backend.

**Screenshots:**

captura de pantalla 2018-10-29 a la s 11 54 09
captura de pantalla 2018-10-29 a la s 12 08 51

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+API_URL=https://target-mvd-api-andres.herokuapp.com/api/v1
+GOOGLE_MAPS_API_KEY=abcdefgh
```
Comment (missx) [inline .env]: Could you add a line at the end of the document and to the ones below that are missing one line?

File: src/components/common/Input/index.js [hunk 2]
```diff
@@ -1,38 +1,45 @@
-import React from 'react';
-import {TextInput} from 'react-native';
+import React, {Component} from 'react';
+import {TextInput, View, Text} from 'react-native';
```
Comment (missx) [inline src/components/common/Input/index.js]: Could you leave a space between words and opening/closing brackets?

File: src/components/common/Select/index.js [hunk 3]
```diff
@@ -1,22 +1,50 @@
...
+class Select extends Component {
+  constructor(props){
+    super(props);
+    this.state = {value: props.input.value};  
+  }
 
-export default Select;
\ No newline at end of file
+  handleOnSelectChange(value) {
+    switch(value) {
```
Comment (Alxzu) [inline src/components/common/Select/index.js]: Can't you make just a parseInt, like `this.props.input.value = parseInt(value)` ?

File: src/components/user/SignInForm/index.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import { Text, View, TouchableOpacity} from 'react-native';
+import { Divider } from 'react-native-elements';
+import Input from '../../common/Input';
+import ActionButton from '../../common/ActionButton';
+import styles from './styles';
+import { white, black } from '../../../constants/styleConstants';
+import * as constraints from '../../../utils/constraints';
+
+const SignInForm = ({ handleSubmit }) => (
+  <View style={styles.container} keyboardShouldPersistTaps={'handled'} onSubmit={handleSubmit}>
```
[the remark this answers is not in the document]
Comment (felipegarcia92) [reply src/components/user/SignInForm/index.js]: I used to think the same thing but now I agree not to do that. 

The reason behind it is that the variable names without the prepending `styles.` might be confused with actual data variables (not styles).

File: src/components/common/Select/index.js (hunk 3 above)
Comment (jdalto) [reply src/components/common/Select/index.js]: I had thought that before doing it, but I did it this way for the case that the value is a string that is not a integer, but in this case I could do that

File: src/containers/SignInScreen/index.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+          </ImageBackground>
+        </View>         
+      </View>
+    );
+  }
+}
+
+const mapState = state => ({
+  authenticated: state.getIn(['session', 'authenticated']),
+  loading: state.getIn(['user', 'loading']),
```
[the remark this answers is not in the document]
Comment (jdalto) [reply src/containers/SignInScreen/index.js]: You are right, the props do not have use, but surely I will use them later on, that is why I leave them

File: .env (hunk 1 above)
Comment (jdalto) [reply .env]: I should have included this file in gitignore, anyway i will do that for the others files

**Pull request merged**